service/history/historyEngine.go

```diff
@@ -908,6 +908,38 @@ Update_History_Loop:
 					ScheduleID:       *wfCancelReqEvent.EventId,
 				})
 
+			case workflow.DecisionTypeSignalExternalWorkflowExecution:
+				e.metricsClient.IncCounter(metrics.HistoryRespondDecisionTaskCompletedScope,
+					metrics.DecisionTypeSignalExternalWorkflowCounter)
+
+				attributes := d.SignalExternalWorkflowExecutionDecisionAttributes
+				if err = validateSignalExternalWorkflowExecutionAttributes(attributes); err != nil {
+					failDecision = true
+					failCause = workflow.DecisionTaskFailedCauseBadSignalWorkflowExecutionAttributes
+					break Process_Decision_Loop
+				}
+
+				foreignInfo, _, err := e.domainCache.GetDomain(attributes.GetDomain())
+				if err != nil {
+					return &workflow.InternalServiceError{
+						Message: fmt.Sprintf("Unable to signal workflow across domain: %v.",
+							attributes.GetDomain())}
+				}
+
+				signalRequestID := uuid.New() // for deduplicate
+				wfSignalReqEvent := msBuilder.AddSignalExternalWorkflowExecutionInitiatedEvent(completedID,
+					signalRequestID, attributes)
+				if wfSignalReqEvent == nil {
+					return &workflow.InternalServiceError{Message: "Unable to add external signal workflow request."}
+				}
+
+				transferTasks = append(transferTasks, &persistence.SignalExecutionTask{
+					TargetDomainID:   foreignInfo.ID,
+					TargetWorkflowID: attributes.GetWorkflowId(),
+					TargetRunID:      common.StringDefault(attributes.RunId),
```

Comment:
Use the generated getter: attributes.GetRunId()

Reply:
Done